File: nonoKing/LeetCode/Cpp/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */
#include <vector>
#include <map>
using namespace std;
// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        if (nums.size() <= 0) return res;
        map<int, int> hashList;
        for (int i = 0; i < nums.size(); ++i)
        {
            int key = target - nums[i];
            map<int, int>::iterator it = hashList.find(key);
            if (it != hashList.end())
            {
                res.push_back(it->second);
                res.push_back(i);
                return res;
            }
            hashList.insert(make_pair(nums[i], i));
        }
        return res;
    }
};
// @lc code=end

File: nonoKing/LeetCode/Python/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i, n in enumerate(nums):
            if target - n in dict:
                return [dict[target - n], i]
            dict[n] = i
# @lc code=end

